Source/JavaScriptCore/b3/air/AirHandleCalleeSaves.cpp

```diff
@@ -58,8 +58,12 @@ void handleCalleeSaves(Code& code)
 
 void handleCalleeSaves(Code& code, RegisterSetBuilder usedCalleeSaves)
 {
-    // We filter to really get the callee saves.
-    usedCalleeSaves.filter(RegisterSetBuilder::calleeSaveRegisters());
+    // We filter to really get the callee saves. On 32-bit, we want the VM callee saves, which are
+    // a subset of the C callee saves.
+    if (is32Bit())
+        usedCalleeSaves.filter(RegisterSetBuilder::vmCalleeSaveRegisters());
```

Comment:
That sounds strange, can you explain why? B3 / Air should be agnostic against VM.